test/unit/lib/challenge.test.ts

```diff
@@ -26,3 +27,173 @@ describe('Challenge Generation', (): void => {
     expect(Buffer.byteLength(challenge, 'base64')).toBe(8)
   })
 })
+
+// Each test generates a random key pair
+describe('Signature Verification', (): void => {
+  let challenge: string
+  let signer: crypto.Signer
+
+  beforeEach((): void => {
+    challenge = 'Crypto Auth service Yay!'
+    signer = crypto.createSign('SHA256')
+
+    signer.update(challenge)
+  })
+
+  it('verifies correct signature - EC Key', (): void => {
+    const keyPair = crypto.generateKeyPairSync('ec', {
+      namedCurve: 'secp256k1', // Allowed by FIDO spec
+      publicKeyEncoding: {
+        type: 'spki', // Key infrasructure
+        format: 'pem' // Encoding format
+      },
+      privateKeyEncoding: {
+        type: 'pkcs8',
+        format: 'pem'
+      }
+    })
+
+    const sign = signer.sign(keyPair.privateKey, 'base64')
+
+    expect(verifySign(challenge, sign, keyPair.publicKey)).toEqual(true)
```

Comment:
Same comment as earlier. Please separate out the assertions from the test action